aea/mail/base.py

```diff
@@ -799,7 +758,74 @@ async def async_wait(self) -> None:
         """
         return await self.in_queue.async_wait()
 
-    def put(self, envelope: Envelope) -> None:
+    async def put(self, envelope: Envelope) -> None:
+        """
+        Schedule an envelope for sending it.
+
+        Notice that the output queue is an asyncio.Queue which uses an event loop
+        running on a different thread than the one used in this function.
+
+        :param envelope: the envelope to be sent.
+        :return: None
+        """
+        await self.out_queue.put(envelope)
+
+
+class Multiplexer(AsyncMultiplexer):
+    """Transit sync multiplexer for compatibility."""
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the connection multiplexer.
+
+        :param connections: a sequence of connections.
+        :param default_connection_index: the index of the connection to use as default.
+                                       | this information is used for envelopes which
+                                       | don't specify any routing context.
+        :param loop: the event loop to run the multiplexer. If None, a new event loop is created.
+        """
+        super().__init__(*args, **kwargs)
+        self._thread_runner = ThreadedAsyncRunner(self._loop)
+        self._sync_lock = threading.Lock()
+        self._thread_was_started = False
+        self._is_connected = False
+
+    def connect(self) -> None:  # type: ignore  # cause overrides coroutine
+        """
+        Connect the multiplexer.
+
+        Synchronously in thread spawned if new loop created.
+        """
+        with self._sync_lock:
+            if not self._loop.is_running():
+                self._thread_runner.start()
+                self._thread_was_started = True
+
+            self._thread_runner.call(super().connect()).result(240)
```

Comment:
not clear what 240 is, a timeout I imagine, make constant with explicit name for better readability